test/runnable_cxx/test19179.d

```diff
@@ -0,0 +1,32 @@
+// EXTRA_CPP_SOURCES: cpp19179.cpp
+
+// https://issues.dlang.org/show_bug.cgi?id=19179
+
+import core.stdc.stdio;
+
+extern(C++) struct SmallStruct { int x = 10, y = 20; }
```

Comment:
Test case should use different types ;-)

Reply:
The types being the same doesn't affect the logic of the compiler.